akka-actor/src/main/scala/akka/io/dns/internal/TcpDnsClient.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.io.dns.internal
+
+import java.net.InetSocketAddress
+
+import akka.AkkaException
+import akka.actor.{ Actor, ActorLogging, ActorRef, Stash }
+import akka.annotation.InternalApi
+import akka.io.Tcp._
+import akka.io.dns.internal.DnsClient.{ Answer, DnsQuestion, Question4 }
+import akka.io.{ IO, Tcp }
+import akka.util.ByteString
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] class TcpDnsClient(ns: InetSocketAddress, answerRecipient: ActorRef) extends Actor with ActorLogging with Stash {
+  import TcpDnsClient._
+
+  import context.system
+
+  override def receive: Receive = idle
+
+  val tcp = IO(Tcp)
+
+  val idle: Receive = {
+    case _: Message ⇒
+      stash()
+      log.debug("Connecting to [{}]", ns)
+      tcp ! Tcp.Connect(ns)
+      context.become(connecting)
+  }
+
+  val connecting: Receive = {
+    case CommandFailed(_: Connect) ⇒
+      log.warning("Failed to connect to [{}]", ns)
+      throw new AkkaException("Connecting failed")
```

Comment:
Isn't just throwing going to lead to it getting logged? Put the `ns` in the exception instead?

Reply:
Probably, but that might not be bad? Weird if connecting to the DNS server fails.

Reply:
Mostly thinking that if we get two errors in the log that is confusing but maybe it doesn't matter.

Reply:
Ah, good one, will remove the `warning` :+1: